daemon/cmd/kube_proxy_replacement.go

```diff
@@ -272,6 +272,27 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		}
 	}
 
+	if option.Config.BPFSocketLBHostnsOnly {
+		if !option.Config.EnableHostReachableServices {
+			option.Config.BPFSocketLBHostnsOnly = false
+			log.Warnf("BPFSocketLBHostnsOnly only takes effect when EnableHostReachableServices is true")
+		} else {
+			found := false
+			if helpers := probesManager.GetHelpers("cgroup_sock_addr"); helpers != nil {
+				if _, ok := helpers["bpf_get_netns_cookie"]; ok {
+					found = true
+				}
+			}
+			if !found {
+				option.Config.BPFSocketLBHostnsOnly = false
+				log.Warn("Without network namespace cookie lookup functionality, BPF datapath " +
+					"cannot distinguish root and non-root namespace, skipping socket-level " +
+					"loadbalancing will not work. Istio routing chains will be missed. " +
+					"Needs kernel version >= 5.7")
```

Comment:
We should also document this in the kernel-requirements table at https://docs.cilium.io/en/latest/operations/system_requirements/#required-kernel-versions-for-advanced-features.

Reply:
Done.